src/GovUk.Education.ExploreEducationStatistics.Admin/Controllers/Api/DataSetFileMetaMigrationController.cs

```diff
@@ -0,0 +1,94 @@
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Admin.Models;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Controllers.Api;
+
+[Route("api")]
+[ApiController]
+[Authorize(Roles = GlobalRoles.RoleNames.BauUser)]
+public class DataSetFileMetaMigrationController(ContentDbContext contentDbContext) : ControllerBase
+{
+    public class DataSetFileMetaMigrationResult
+    {
+        public bool IsDryRun;
+        public int Processed;
+    }
+
+    [HttpPut("bau/migrate-datasetfilemeta")]
+    public async Task<DataSetFileMetaMigrationResult> MigrationDataSetFileMeta(
+        [FromQuery] bool dryRun = true,
+        [FromQuery] int? num = null,
```

Comment:
If the intended purpose of this parameter is to manage a large number of items as separate calls, should there not also be some ordering and an implementation of `Skip` prior to `Take`? Appreciate the `Where` clauses would exclude records which have already been populated, so just curious on the approach as it seems like quite a manual process for a post-deployment task (i.e. you'd need to hit the endpoint a variable number of times per environment).